From 4c18c56ae4096596b08458db940981ba6d23513a Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 20 Jul 2021 20:37:45 +0300
Subject: [PATCH] [SW-0] greco: increase drain timeout

This patch increases AXI DRAIN timeout of PSOC and PCIE to 0x4000
(16384 cycles) as recommended by hardware team. The drain should
be high enough to ensure that the traffic is observed by the SoC.

Change-Id: If804bc1c9928016f080f097ec1346bdf1b0836fa
---

diff --git a/arch/greco/arc_elf32/common.h b/arch/greco/arc_elf32/common.h
index 6716ae0..76c9933 100644
--- a/arch/greco/arc_elf32/common.h
+++ b/arch/greco/arc_elf32/common.h
@@ -51,6 +51,6 @@
 #define PCIE_WAIT_TIMEOUT_US	(5 * USEC_PER_SEC)
 #define PCIE_EXT_LD_TIMEOUT_US	576
 
-#define PSOC_AXI_DRAIN_TIMEOUT		0x3E8 /* 1000 cycles */
+#define PSOC_AXI_DRAIN_TIMEOUT		0x4000 /* 16K cycles */
 
 #endif /* COMMON_GRECO_ARC_H_ */
diff --git a/arch/greco/arm/common.h b/arch/greco/arm/common.h
index 8ea4d58..58b7d36 100644
--- a/arch/greco/arm/common.h
+++ b/arch/greco/arm/common.h
@@ -83,6 +83,6 @@
 #define PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3_CONF	0x5555
 #define PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4_CONF	0xFF
 
-#define PSOC_AXI_DRAIN_TIMEOUT				0x3E8 /* 1000 cycles */
+#define PSOC_AXI_DRAIN_TIMEOUT				0x4000 /* 16K cycles */
 #define PCIE_AXI_DRAIN_TIMEOUT				PSOC_AXI_DRAIN_TIMEOUT
 #endif /* COMMON_GRECO_H_ */
